src/SimulationMap.cuh

```diff
@@ -12,25 +12,87 @@ class SimulationMap
 {
 public:
     /**
-     * Creates a `SimulationMap` object
+     * Creates a `SimulationMap` object and a grid of nodes
      *
-     * This function isn't implemented yet, neither it's ready to be implemented, so the description stays
-     * empty for now
+     * @param polyhedron The polyhedron in simulation
      */
-    __device__ SimulationMap(...);
+    __device__ SimulationMap(Polyhedron *polyhedron);
 
     /// Forbids copying `SimulationMap` objects
     __host__ __device__ SimulationMap(const SimulationMap &) = delete;
 
     /// Destructs a `SimulationMap` object
     __device__ ~SimulationMap();
 
+
+    /**
+     * Returns coordinates of neighbor node with or without projection on polyhedron
+     *
+     * @param current_node_id Index of the node whose neighbor is searched
+     * @param top_direction Direction vector from current node to its top neighbor
+     * @param angle Angle between the top neighbor node and the neighbor node whose index is searched
+     *              relative to current node, clockwise is positive direction
+     * @param do_projection If `true`, counted coordinates will be projected on polyhedron, otherwise they will not
+     *
+     * @returns Coordinates of neighbor node projected on polyhedron if `do_projection` is `true`,
+     *          coordinates of neighbor node without projection on polyhedron otherwise
+     */
+    __device__ SpacePoint count_neighbor_node_coordinates(int current_node_id, SpacePoint top_direction, double angle,
+                                                          bool do_projection) const;
+
+
+    /**
+     * Finds the nearest node to the given point
+     *
+     * @param point_coordinates Coordinates of the point that the closest node is searched to
+     *
+     * @returns The index of the found node in `nodes` array
+     */
+    __device__ int find_nearest_node_to_point(SpacePoint point_coordinates) const;
```

Comment:
The function's name is really bad because there is a function called `find_nearest_mapnode` which does nearly the same thing.

The first problem is that despite they are giving always the same result, their names significantly differ, which may create a feeling that `find_nearest_mapnode` does not find a mapnode nearest **to a point**, or that `find_nearest_node_to_point` finds some node which is not a mapnode.

The second problem is that the difference between these two functions is neither obvious from their names nor even documented!